--- a/net/minecraft/util/StringUtils.java
+++ b/net/minecraft/util/StringUtils.java
@@ -2,18 +2,34 @@
 
 import java.util.regex.Pattern;
 import javax.annotation.Nullable;
+import java.text.NumberFormat;
 import net.minecraftforge.api.distmarker.Dist;
 import net.minecraftforge.api.distmarker.OnlyIn;
 
 public class StringUtils {
    private static final Pattern field_76339_a = Pattern.compile("(?i)\\u00A7[0-9A-FK-OR]");
+   private static final NumberFormat numberFormat = NumberFormat.getIntegerInstance();
+   private static final NumberFormat numberFormat00 = NumberFormat.getIntegerInstance();
+   private static final NumberFormat percentFormat = NumberFormat.getPercentInstance();
+   static {
+      numberFormat00.setMinimumIntegerDigits(2);
+      percentFormat.setMaximumFractionDigits(0);
+   }
 
+   public static String formatInteger(long value) {
+      return numberFormat.format(value);
+   }
+
+   public static String formatPercent(float ratio) {
+      return percentFormat.format(ratio);
+   }
+
    @OnlyIn(Dist.CLIENT)
    public static String func_76337_a(int p_76337_0_) {
       int i = p_76337_0_ / 20;
       int j = i / 60;
       i = i % 60;
-      return i < 10 ? j + ":0" + i : j + ":" + i;
+      return numberFormat.format(j) + ":" + numberFormat00.format(i);
    }
 
    @OnlyIn(Dist.CLIENT)
