--- a/net/minecraft/client/gui/screen/WorldLoadProgressScreen.java
+++ b/net/minecraft/client/gui/screen/WorldLoadProgressScreen.java
@@ -4,6 +4,7 @@
 import it.unimi.dsi.fastutil.objects.Object2IntOpenHashMap;
 import net.minecraft.client.gui.chat.NarratorChatListener;
 import net.minecraft.client.resources.I18n;
+import net.minecraft.util.StringUtils;
 import net.minecraft.util.Util;
 import net.minecraft.util.math.MathHelper;
 import net.minecraft.util.text.TranslationTextComponent;
@@ -48,7 +49,7 @@
 
    public void render(int p_render_1_, int p_render_2_, float p_render_3_) {
       this.renderBackground();
-      String s = MathHelper.func_76125_a(this.field_213040_a.func_219524_e(), 0, 100) + "%";
+      String s = StringUtils.formatPercent(MathHelper.func_76125_a(this.field_213040_a.func_219524_e(), 0, 100) / 100.0f);
       long i = Util.func_211177_b();
       if (i - this.field_213041_b > 2000L) {
          this.field_213041_b = i;
